lib/system/nuttx/irq.c

```diff
@@ -45,7 +45,7 @@ static int metal_cntr_irq_handler(int irq, void *context, void *data)
 
 	/* context == NULL mean unregister */
 	irqchain_detach(irq, metal_cntr_irq_handler, data);
-	sched_kfree(data);
+	kmm_free(data);
```

Comment:
As metal_allocate_memory is used you should use metal_free_memory ( which is relying on kmm_free)

Reply:
Good finding!